### ISSUE TITLE
test size

### ISSUE THREAD

**Issue by tuvovan** [claroche-r/DMBSR#3]
So the size of the image in the test phase is fixed to 256 by 256? 
does it work with any bigger size, says in real application?

**Comment by claroche-r** [comment]
Hi tuvovan,

In the paper, we only used 256x256 patches but in practice, I tried bigger ones. The network is fully convolutional so there are no problems. 
In practice, you have two options with big images:
- Directly apply the model to your bigger images (at some point, you will have memory issues but at test time it is not as costly).
- Use tilling. i.e. You split your image into patches (generally overlapping), apply the model to the patches, and reconstruct the image.

Hope this answer your question.

Best,
Charles

**Comment by tuvovan** [comment]
I see. However as the blur is non-uniform, does it still work if training data and testing data have different size?

**Comment by claroche-r** [comment]
It could be a problem if the model is blind, in our case the model is non-blind and the deblurring is done by a gradient descent on the blur operator (see [Equation  (13)](https://openaccess.thecvf.com/content/WACV2023/papers/Laroche_Deep_Model-Based_Super-Resolution_With_Non-Uniform_Blur_WACV_2023_paper.pdf) of the paper with the explanations below). The relation between the blur and the model is explicit, so image size is not a problem.

Keep in mind that if you do tiling, you have to tile both the degraded image and the kernel masks ("kmap" in the code).

**Comment by tuvovan** [comment]
thank you very much. just one final question, I guess ADMM with inverse of non-uniform blur kernel (say applying fft on each patch of the image with corresponding kernel) still works right? I mean in term of backpropagation...

thank you!

**Comment by claroche-r** [comment]
I am not sure to completely understand your question. You cannot invert spatially varying blur using ADMM since the proximal operator of the data fitting term is not tractable. Indeed, you have to invert the matrix:
$$(H^TH + c Id)$$
with H the blur operator. This cannot be done directly for spatially varying blur, even in Fourier (the Fourier transform of a non-uniform blur is not diagonal since it is no longer a convolution). The main idea of our [paper](https://openaccess.thecvf.com/content/WACV2023/papers/Laroche_Deep_Model-Based_Super-Resolution_With_Non-Uniform_Blur_WACV_2023_paper.pdf) is to bypass this inversion using linearized-ADMM.

**Comment by tuvovan** [comment]
yeah I understand that we can not invert the whole image. but just in case we know the blur kernel for each corresponding image patch, we can always do the invert using ADMM on each patch. Please correct me if I'm worng.! Thank you! 👍🏿

**Comment by claroche-r** [comment]
Ok, I understand. If you have a separate blur for each patch you can do that. However, it might introduce artifacts at the border of the different patches (see the results of USRNet Figure 7 in our paper, that is exactly what we did).

Moreover, if you have a linear transition between two blur kernels (which is what happens in practice) it is impossible to do that since you will have a different kernel for each pixel (or line/column).
